Sources/eppo/RuleEvaluator.swift

```diff
@@ -175,95 +240,131 @@ public class FlagEvaluator {
+        
         // Safely unwrap attributeValue for further use
         guard let value = attributeValue else {
             // Handle the nil case, perhaps throw an error or return a default value
             return false
         }
-
-        do {
-            switch condition.operator {
-            case .greaterThanEqual, .greaterThan, .lessThanEqual, .lessThan:
-                do {
-                    let valueStr = try? value.getStringValue()
-                    let conditionValueStr = try? condition.value.getStringValue()
-                    if let valueVersion = valueStr.flatMap(Semver.init), let conditionVersion = conditionValueStr.flatMap(Semver.init) {
-                        // If both strings are valid Semver strings, perform a Semver comparison
-                        switch condition.operator {
-                        case .greaterThanEqual:
-                            return valueVersion >= conditionVersion
-                        case .greaterThan:
-                            return valueVersion > conditionVersion
-                        case .lessThanEqual:
-                            return valueVersion <= conditionVersion
-                        case .lessThan:
-                            return valueVersion < conditionVersion
-                        default:
-                            throw Errors.UnexpectedValue
-                        }
-                    } else {
-                        // If either string is not a valid Semver, fall back to double comparison
-                        let valueDouble = try value.getDoubleValue()
-                        let conditionDouble = try condition.value.getDoubleValue()
-                        switch condition.operator {
-                        case .greaterThanEqual:
-                            return valueDouble >= conditionDouble
-                        case .greaterThan:
-                            return valueDouble > conditionDouble
-                        case .lessThanEqual:
-                            return valueDouble <= conditionDouble
-                        case .lessThan:
-                            return valueDouble < conditionDouble
-                        default:
-                            throw Errors.UnexpectedValue
-                        }
-                    }
-                } catch {
-                    // If stringValue() or doubleValue() throws, or Semver creation fails
+        
+        switch condition.operator {
+        case .greaterThanEqual, .greaterThan, .lessThanEqual, .lessThan:
+            let valueStr = try? value.getStringValue()
+            
+            // If the config is obfuscated, we need to unobfuscate the condition value
+            var conditionValueStr: String? = try? condition.value.getStringValue()
+            if isConfigObfuscated,
+               let cvs = conditionValueStr,
+               let decoded = base64Decode(cvs) {
+                conditionValueStr = decoded
+            }
```

Comment:
should this `}` and the line above be indented more? Conditional evaluation over multiple lines is a bit mind-bending 😛 

Is the reason for the code block because we're using the result from the previous binding?